pints/_mcmc/_dual_averaging.py

```diff
@@ -0,0 +1,256 @@
+#
+# Dual Averaging step size and mass matrix adaption method for NUTS and HMC
+# samplers
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+import numpy as np
+
+
+class DualAveragingAdaption:
+    r"""
+    Implements a Dual Averaging scheme to adapt the step size ``epsilon``, as
```

Comment:
Can we do a 1 liner, followed by a new paragraph (pep8!) The one-liner can be used by e.g. IDEs to show a single line explanation of what this class does

Reply:
Probably there should just be a whole new first line that says this is a thing for NUTS & HMC ?

I've no idea after reading the below stuff :D